pkgs/racket-doc/scribblings/reference/treelists.scrbl

```diff
@@ -455,17 +455,16 @@ If @racket[s] is infinite, this function does not terminate.
 @history[#:added "8.15.0.6"]}
 
 @deftogether[(
-@defform[(for/treelist (for-clause ...) body-or-break ... body)]
-@defform[(for*/treelist (for-clause ...) body-or-break ... body)]
+@defform[(for/treelist maybe-length (for-clause ...) body-or-break ... body)]
+@defform[(for*/treelist maybe-length (for-clause ...) body-or-break ... body)]
 )]{
 
-Like @racket[for/list] and @racket[for*/list], but generating
+Like @racket[for/vector] and @racket[for*/vector], but generating
 @tech{treelists}.
 
 @examples[
 #:eval the-eval
-(for/treelist ([i (in-range 10)])
-  i)
+(for/treelist #:length 15 #:fill 'a ([i (in-range 10)]) i)
```

Comment:
I think we should add new examples, rather than modifying the existing example. The original form is probably going to be the common use, still, and we should include that.